test/algorithms/excited_state_solvers/test_excited_states_solvers.py

```diff
@@ -151,6 +151,36 @@ def filter_criterion(eigenstate, eigenvalue, aux_values):
         for idx, energy in enumerate(self.reference_energies):
             self.assertAlmostEqual(computed_energies[idx], energy, places=4)
 
+    def test_custom_filter_criterion(self):
+        """Test NumPyEigenSolverFactory with ExcitedStatesEigensolver and a custom filter criterion"""
+
+        expected_spin = 0
```

Comment:
Maybe we should instead set-up a non-singlet example for this test. For example `BeH` which is a doublet (`spin = 1`)